### ISSUE TITLE
Check failed: proto.SerializeToOstream(&output)

### ISSUE THREAD

**Issue by whjxnyzh** [facebook/C3D#13]
Hi @dutran 
I run your example ucf101_finetuning.sh

when iteration 10000, error occurs
```
F0606 09:01:48.148102 23862 io.cpp:72] Check failed: proto.SerializeToOstream(&output)
*** Check failure stack trace: ***
    @     0x7fb7ee823b5d  google::LogMessage::Fail()
    @     0x7fb7ee827b77  google::LogMessage::SendToLog()
    @     0x7fb7ee8259f9  google::LogMessage::Flush()
    @     0x7fb7ee825cfd  google::LogMessageFatal::~LogMessageFatal()
    @           0x4508eb  caffe::WriteProtoToBinaryFile()
    @           0x43c4bf  caffe::Solver<>::Snapshot()
    @           0x43ecbe  caffe::Solver<>::Solve()
    @           0x41485d  main
    @       0x3d1f01ed1d  (unknown)
    @           0x414579  (unknown)
ucf101_finetuning.sh: line 1: 23862 Aborted                 (core dumped) GLOG_logtostderr=1 ../../build/tools/finetune_net.bin c3d_ucf101_finetuning_solver.prototxt ../c3d_feature_extraction/conv3d_deepnetA_sport1m_iter_1900000
```
but there are no errors in iteration 1000 to 9000. I think it is strange, my GPU is tesla k40
```
[nlpr4@bnode01 c3d_finetuning]$ ls
c3d_ucf101_finetune_whole_iter_1000              c3d_ucf101_finetune_whole_iter_8000
c3d_ucf101_finetune_whole_iter_10000             c3d_ucf101_finetune_whole_iter_8000.solverstate
c3d_ucf101_finetune_whole_iter_1000.solverstate  c3d_ucf101_finetune_whole_iter_9000
c3d_ucf101_finetune_whole_iter_2000              c3d_ucf101_finetune_whole_iter_9000.solverstate
c3d_ucf101_finetune_whole_iter_2000.solverstate  c3d_ucf101_finetuning_solver.prototxt
c3d_ucf101_finetune_whole_iter_3000              c3d_ucf101_finetuning_test.prototxt
c3d_ucf101_finetune_whole_iter_3000.solverstate  c3d_ucf101_finetuning_train.prototxt
c3d_ucf101_finetune_whole_iter_4000              frame_to_avi.py
c3d_ucf101_finetune_whole_iter_4000.solverstate  myscript
c3d_ucf101_finetune_whole_iter_5000              readme.md
c3d_ucf101_finetune_whole_iter_5000.solverstate  read_video.py
c3d_ucf101_finetune_whole_iter_6000              train01_16_128_171_mean.binaryproto
c3d_ucf101_finetune_whole_iter_6000.solverstate  train_01.lst
c3d_ucf101_finetune_whole_iter_7000              ucf101_finetuning.sh
c3d_ucf101_finetune_whole_iter_7000.solverstate  ucf101_testing.sh
```

**Comment by dutran** [comment]
It looks strange to me too. It seems that your solver failed when it was trying to save a snapshot. Can you check if you have any more space on disk?

**Comment by whjxnyzh** [comment]
oh, I just resume traing from iteration 9000,
```
[nlpr4@bnode01 c3d_finetuning]$ GLOG_logtostderr=1 ../../build/tools/train_net.bin \
> c3d_ucf101_finetuning_solver.prototxt c3d_ucf101_finetune_whole_iter_9000.solverstate
``` 
this time it worked...... 
```
I0606 10:37:48.255017 64324 solver.cpp:237] Iteration 10000, lr = 1e-05
I0606 10:37:48.255482 64324 solver.cpp:87] Iteration 10000, loss = 0.0224443
I0606 10:37:48.869892 64324 solver.cpp:159] Snapshotting to c3d_ucf101_finetune_whole_iter_                                                          10000
I0606 10:37:50.094324 64324 solver.cpp:166] Snapshotting solver state to c3d_ucf101_finetun                                                          e_whole_iter_10000.solverstate
I0606 10:39:26.923598 64324 solver.cpp:237] Iteration 10020, lr = 1e-05
I0606 10:39:26.924058 64324 solver.cpp:87] Iteration 10020, loss = 0.0513535
I0606 10:41:03.297607 64324 solver.cpp:237] Iteration 10040, lr = 1e-05
I0606 10:41:03.298079 64324 solver.cpp:87] Iteration 10040, loss = 0.0078975

```
I really don't know what happened last time. any way. it is working now. thank you

**Comment by whjxnyzh** [comment]
i will close this issue

**Comment by dutran** [comment]
Glad to hear!

**Comment by yzloveyz** [comment]
Hi @whjxnyzh , thanks for posting out your training process. I also fine-tuning the UCF101 model these days, but for iteration 10000, I still got 0.05 to 0.2 loss, much bigger than your loss number. So I was wondering if there is any trick during training? Like you changed your lr to 0.00001, or something else? Thanks. BTW, if you could share your iteration 20000 model, I would greatly appreciated.
